src/main/webapp/app/overview/course-overview.component.ts

```diff
@@ -194,6 +204,94 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
         await this.initAfterCourseLoad();
         this.sidebarItems = this.getSidebarItems();
         this.courseActionItems = this.getCourseActionItems();
+        this.updateVisibility(window.innerHeight);
+        this.updateMenuOffset();
+    }
+
+    /** Listen window resizement event by height */
+    @HostListener('window: resize', ['$event'])
+    onResize() {
+        this.dropdownOpen = false;
+        this.dropdownClickNumber = 0;
+        this.updateVisibility(window.innerHeight);
+        this.updateMenuOffset();
+    }
+
+    /** Listen click event whether on outside of the menu or one of the items in the menu to close the dropdown menu */
+    @HostListener('document: click', ['$event'])
+    onClickCloseDropdownMenu() {
+        if (this.dropdownOpen) {
+            this.dropdownClickNumber += 1;
+            if (this.dropdownClickNumber === 2) {
+                this.dropdownOpen = false;
+                this.dropdownClickNumber = 0;
+            }
+        }
+    }
+
+    /** Update sidebar item's hidden property based on the window height to display three-dots */
+    updateVisibility(height: number) {
+        const thresholds: number[] = [];
+        const itemHeight: number = 35;
+        let threshold = this.calculateThreshold();
```

Comment:
For me `threshold` does not contain a lot information, I would appreciate a more descriptive variable name (in my opinion it doesn't matter if it is a long name)